crates/proto/go/src/download.rs

```diff
@@ -0,0 +1,96 @@
+use crate::platform::GoArch;
+use crate::GoLanguage;
+use log::debug;
+use proto_core::{
+    async_trait, color, download_from_url, Describable, Downloadable, ProtoError, Resolvable,
+};
+use std::env::consts;
+use std::path::{Path, PathBuf};
+
+#[cfg(target_os = "macos")]
+pub fn get_archive_file_path(version: &str) -> Result<String, ProtoError> {
+    let arch = GoArch::from_os_arch()?;
+
+    if !matches!(arch, GoArch::Amd64 | GoArch::Arm64) {
```

Comment:
are all these matches conditions accurate?

Reply:
Pretty sure. I ran a script to find the packages available:

```
✅ darwin amd64 tar.gz
✅ darwin arm64 tar.gz
✅ linux 386 tar.gz
✅ linux amd64 tar.gz
✅ linux arm64 tar.gz
✅ linux armv6l tar.gz
✅ linux s390x tar.gz
✅ windows 386 zip
✅ windows amd64 zip
✅ windows arm64 zip
```

Reply:
dope